kotlinx-collections-immutable/src/main/kotlin/kotlinx/collections/immutable/implementations/immutableMap/TrieNode.kt

```diff
@@ -7,70 +7,123 @@ internal const val MAX_BRANCHING_FACTOR_MINUS_ONE = MAX_BRANCHING_FACTOR - 1
 internal const val ENTRY_SIZE = 2
 internal const val MAX_SHIFT = 30
 
+/**
+ * Gets trie index segment of the specified [index] at the level specified by [shift].
+ *
+ * `shift` equal to zero corresponds to the root level.
+ * For each lower level `shift` increments by [LOG_MAX_BRANCHING_FACTOR].
+ */
+internal fun indexSegment(index: Int, shift: Int): Int =
```

Comment:
Is it worth to name this `indexSegmentMapTrie` (and correspondingly `indexSegmentSetTrie`) in order not to confuse them accidentally with auto completion?

Reply:
There is also `indexSegment` in persistent list implementation, and all these functions have identical implementations. 
One of the solutions is to move this function to a common util file.

I don't think it worths renaming, because accidentally selecting inappropriate implementation with auto completion writes fully qualified name of the selected function.

Reply:
They are currently same, but they use different constants. So any of these constants can be changed independently, and in that case confusing the function would lead to a wrong result.